ui/js/filters.js

```diff
@@ -72,6 +72,12 @@ treeherder.filter('initials', function() {
         var words = str.split(' ');
         var first = words[0].replace(/[^A-Z]/gi, '')[0];
         var last = words.slice(-1)[0].replace(/[^A-Z]/gi, '')[0];
+        if (!first) {
+            first = "";
+        }
+        if (!last) {
+            last = "";
+        }
```

Comment:
So this looks ok, but on inspection I don't think a better solution is that hard:

```js
initials = _filter.(_.map(words, function(word) { return word.replace(/[^A-Z]/gi, '')[0]));
```

Then you could take the first and last items from that, if there are more than two values. This would correctly abbreviate "Kan-Ru Chen (<unicode>)" to KC